### PR TITLE
detect-sameip: convert unittests to FAIL/PASS APIs

### PR DESCRIPTION
**Pull request by TheKharleeci**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to redmine ticket:
https://redmine.openinfosecfoundation.org/issues/4057

Ticket: #4057

Previous Pr: #6559

Describe changes:

- Update Detect-sameip unittests to use FAIL/PASS APIs
- Update Copyright Year

### CODE REVIEW

File: src/detect-sameip.c [hunk 1]
```diff
@@ -143,45 +142,28 @@ static int DetectSameipSigTest01(void)
...
-    if (PacketAlertCheck(p2, 1) != 0) {
-        printf("sid 2 alerted, but should not have: ");
-        goto cleanup;
-    }
-
-    result = 1;
-
-cleanup:
-    SigGroupCleanup(de_ctx);
-    SigCleanSignatures(de_ctx);
```
Comment (inashivb) [inline src/detect-sameip.c:178]: Q: Why did we remove these cleanup functions? Are they covered elsewhere?

**Comment by inashivb** [comment]
@TheKharleeci this looks nice. Great job! :)
I have one question inline though.

File: src/detect-sameip.c (hunk 1 above)
Comment (TheKharleeci) [reply src/detect-sameip.c:178]: DetectEngineCtxFree will call those cleanup functions in suricata/src/detect-engine.c

Comment (jufajardini) [reply src/detect-sameip.c:178]: I suggested this, following a feedback received by Victor when I was working with one such a task, last year. :P

In the current history: https://github.com/OISF/suricata/pull/6534#discussion_r739618037

Comment (inashivb) [reply src/detect-sameip.c:178]: Awesome! :)

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #6582, thanks!

**Pull request closed without merging**